src/swap/components/SwapSettings.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Children, cloneElement, isValidElement, useMemo } from 'react';
+import { cn } from '../../styles/theme';
+import { useIcon } from '../../wallet/hooks/useIcon';
+import type { SwapSettingsReact } from '../types';
+import { SwapSettingsSlippageLayout } from './SwapSettingsSlippageLayout';
+// import { useBreakpoints } from '../../useBreakpoints';
+
+export function SwapSettings({
+  children,
+  className,
+  icon = 'swapSettings',
+  text = '',
+}: SwapSettingsReact) {
+  // TODO: Add SwapSettingsBottomSheet component for mobile breakpoint.
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const childrenArray = useMemo(() => {
+    return Children.toArray(children).map((child) => {
+      if (isValidElement(child)) {
+        // @ts-ignore
+        return cloneElement(child);
+      }
+      return child;
+    });
+  }, [children]);
+
+  const handleToggle = useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen]);
+
+  const handleClickOutsideComponent = useCallback((event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(event.target as Node)
+    ) {
+      setIsOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutsideComponent);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutsideComponent);
+    };
+  }, [handleClickOutsideComponent]);
+
+  const iconSvg = useIcon({ icon });
+
+  return (
+    <div
+      className={cn(
+        'flex w-full items-center justify-end space-x-1',
+        className,
+      )}
+      data-testid="ockSwapSettings_Settings"
+    >
+      <span className="font-sans text-base leading-normal">{text}</span>
```

Comment:
Should the `<span>` stays there if the `text` is empty?

Reply:
Updated to:

```suggestion
            {text && (
        <span className="font-sans text-base leading-normal">{text}</span>
      )}
```